File: two pointers/lc42.py
def trap(height):

    maxn = height.index(max(height))

    lmax = 0
    sum = 0

    for i in range(lmax, maxn):
        if height[i]>height[lmax]:
            lmax = i
        sum += height[lmax] - height[i]
    
    rmax = len(height)-1

    for i in range(rmax, maxn, -1):
        if height[i]>height[rmax]:
            rmax = i
        sum += height[rmax] - height[i]

    return sum


print(trap([4,2,0,3,2,5]))

File: phase 1/01 - two pointers/lc42.js
/**
 * @param {number[]} height
 * @return {number}
 */
var trap = function(height) {
    let max = 0

    for (let i = 0; i<height.length; i++){
        if(height[i]> height[max]){
            max = i
        }
    }

    let sum = 0
    let lmax = 0

    for (let i = lmax; i<max; i++){
        if (height[i]>height[lmax]){
            lmax = i
        }
        sum += height[lmax] - height[i]
    }

    let rmax = height.length-1
    for (let i = rmax; i>max; i--){
        if (height[i]>height[rmax]){
            rmax = i
        }
        sum +=height[rmax] - height[i]
    }

    return sum
};

console.log(trap([4,2,0,3,2,5]));